### PR TITLE
[updated] TradeShip visit using ScheduleEvents

### PR DESCRIPTION
**Pull request by sboigelot**
feature #1075 

using the new schedule events from #776

This PR allow trader to come each 5 minutes as soon as the player build a landing pad. The time between visit will be variable and implemented in a separate PR.

Also add a option "DevModeOnly" to the ContextMenuAction of furnitures, to allow contextual menu action to only be visible in developer mode. This allow to still call in a trader from a landing pad while testing, but not while playing.

Created a TradeController to centralize the code about trading.

### CODE REVIEW

**Comment by bjubes** [comment]
if the devModeOnly tag is omitted from the XML, does it default to false?

**Comment by sboigelot** [comment]
@bjubes no, I don't think the reader can handle it for now and I didn't wanted to change too much as long as the discussion about XML vs JSON is still hot: #1034

**Comment by bjubes** [comment]
well GetSettingAsBool is passed a default of false in your code, so if omitting tags is supported then it will default to false. If omitting tags isn't supported then that's a different problem entirely.

**Comment by sboigelot** [comment]
@bjubes The context menu action builder in the furniture support it, the Xml reader of furniture prototype doesn't

**Comment by vogonistic** [comment]
This PR has conflicts.

**Comment by vogonistic** [comment]
I really think that DevModeOnly should have a default value of false. If the parser can't handle it, then we ought to create an issue to fix that, but it wouldn't be good if everyone needs to put in a tag for something that should only apply to pretty much no contextual actions.

**Comment by sboigelot** [comment]
Merge conflict fixed + @vogonistic & @bjubes now the xml parsing of contextual menu action allow the developer to forget the DevModeOnly, the default value will be false.

File: Assets/Scripts/Models/Furniture.cs [hunk 1]
```diff
@@ -766,12 +767,16 @@ public IEnumerable<ContextMenuAction> GetContextMenuActions(ContextMenu contextM
 
         foreach (ContextMenuLuaAction contextMenuLuaAction in contextMenuLuaActions)
         {
-            yield return new ContextMenuAction
+            if (!contextMenuLuaAction.DevModeOnly ||
+                Settings.GetSettingAsBool("DialogBoxSettings_developerModeToggle", false))
```
Comment (vogonistic) [inline Assets/Scripts/Models/Furniture.cs]: This is a compile error now. The new prototype is just `GetSetting` and it infers the type form the default, so you can remove `AsBool`.

Comment (sboigelot) [inline Assets/Scripts/Models/Furniture.cs]: I will update the branch

Comment (sboigelot) [inline Assets/Scripts/Models/Furniture.cs]: done :)

**Comment by sboigelot** [comment]
@vogonistic I updated the code to the new API of the Settings

**Comment by bjubes** [comment]
Merged!

**Pull request merged**